WireAPI/Sources/WireAPI/Network/NetworkService/NetworkService.swift

```diff
@@ -122,27 +124,23 @@ extension NetworkService: URLSessionDataDelegate {
     public func urlSession(
         _ session: URLSession,
         task: URLSessionTask,
-        didReceive challenge: URLAuthenticationChallenge,
-        completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void
-    ) {
-        print("task did receive challenge")
-
+        didReceive challenge:
+        URLAuthenticationChallenge
+    ) async -> (URLSession.AuthChallengeDisposition, URLCredential?) {
         let protectionSpace = challenge.protectionSpace
 
-        guard protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust else {
-            completionHandler(.performDefaultHandling, challenge.proposedCredential)
-            return
-        }
+        if protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust {
+            guard let trust = challenge.protectionSpace.serverTrust else { fatalError("Missing server trust") }
```

Comment:
If `serverTrust` is nil this is an Apple breaking its contract. Currently we pass this from obj-c to swift and it is implicitly force unwrapped and not crashing. Therefore, I choosing to fatal in the migration.

Reply:
Maybe worth a comment here explaining the fatal error.